editoast/src/schema/rolling_stock.rs

```diff
@@ -0,0 +1,173 @@
+#[diesel(table_name = osrd_infra_rollingstock)]
+pub struct RollingStock {
+    pub id: i64,
+    name: String,
+    version: String,
+    effort_curves: JsonValue,
+    base_power_class: String,
+    length: f64,
+    max_speed: f64,
+    startup_time: f64,
+    startup_acceleration: f64,
+    comfort_acceleration: f64,
+    gamma: JsonValue,
+    inertia_coefficient: f64,
+    features: Vec<String>,
+    mass: f64,
+    rolling_resistance: JsonValue,
+    loading_gauge: String,
+    metadata: JsonValue,
+    power_restrictions: Option<JsonValue>,
+}
+
+impl RollingStock {
+    pub fn retrieve(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<RollingStock> {
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        match dsl::osrd_infra_rollingstock
+            .find(rolling_stock_id)
+            .first(&mut conn)
+        {
+            Ok(rolling_stock) => Ok(rolling_stock),
+            Err(DieselError::NotFound) => Err(RollingStockError::NotFound(rolling_stock_id).into()),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    pub async fn create(
+        db_pool: Data<DbPool>,
+        rolling_stock_data: RollingStockForm,
+    ) -> Result<RollingStock> {
+        block::<_, Result<_>>(move || {
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match diesel::insert_into(dsl::osrd_infra_rollingstock)
+                .values(&rolling_stock_data)
+                .get_result::<RollingStock>(&mut conn)
+            {
+                Ok(rolling_stock) => Ok(rolling_stock),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    pub async fn delete(db_pool: Data<DbPool>, rolling_stock_id: i64) -> Result<()> {
+        block::<_, Result<_>>(move || {
+            use crate::tables::osrd_infra_rollingstock::dsl::*;
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match diesel::delete(osrd_infra_rollingstock.filter(id.eq(rolling_stock_id)))
+                .execute(&mut conn)
+            {
+                Ok(1) => Ok(()),
+                Ok(_) => Err(RollingStockError::NotFound(rolling_stock_id).into()),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "rollingstocks", context = "Self::context")]
+enum RollingStockError {
+    #[error("Rolling stock '{0}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound(i64),
+}
+
+impl RollingStockError {
+    fn context(&self) -> Map<String, JsonValue> {
+        match self {
+            RollingStockError::NotFound(rolling_stock_id) => json!({
+                "rolling_stock_id": rolling_stock_id,
+            })
+            .as_object()
+            .cloned()
+            .unwrap(),
+        }
+    }
+}
```

Comment:
You don't need this anymore checkout this PR: https://github.com/DGEXSolutions/osrd/pull/3389

TLDR: You just have to adapt your enum with named variant:

```rust
[derive(Debug, Error, EditoastError)]
#[editoast_error(base_id = "rollingstocks")]
enum RollingStockError {
    #[error("Rolling stock '{rolling_stock_id}', could not be found")]
    #[editoast_error(status = 404)]
    NotFound{rolling_stock_id: i64},
}
```